components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/LegacyJobOrchestrator.cs

```diff
@@ -440,32 +458,72 @@ public class LegacyJobOrchestrator : IJobOrchestrator, IPublisherConfigServices,
             _agentConfig.TriggerConfigUpdate(this, new EventArgs());
         }
 
+        private void ValidateRequest(PublishedNodesEntryModel request) {
+            var requestList = new List<PublishedNodesEntryModel> { request };
+            var requestJson = _jsonSerializer.SerializeToString(requestList);
+
+            // This will throw SerializerException if values of request fields are not conformant.
+            DeserializePublishedNodes(requestJson);
+        }
+
+        /// <summary>
+        /// Persist _publishedNodesEntries to published nodes file.
+        /// </summary>
+        private void PersistPublishedNodes() {
```

Comment:
>PersistPublishedNodes [](http://example.com/codeflow?start=21&length=21)

I think we could refresh the _lastKnownFileHash in case the file was successfully written.

Reply:
Good catch, thanks.

Reply:
Added.